src/Composer/DefinitionMutatorInterface.php

```diff
@@ -0,0 +1,24 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Nusje2000\ComposerMonolith\Composer;
+
+interface DefinitionMutatorInterface
+{
+    public function setDependency(string $dependency, string $versionConstraint): void;
+
+    public function removeDependency(string $dependency): void;
+
+    public function setDevDependency(string $dependency, string $versionConstraint): void;
+
+    public function removeDevDependency(string $dependency): void;
+
+    public function setReplace(string $name, string $version): void;
+
+    public function removeReplace(string $name): void;
+
+    public function isMutated(): bool;
+
+    public function save(): void;
```

Comment:
What if save fails for some reason (file permissions etc)?

Reply:
The only check that is done is if the path exists. File permissions are not handled currently.

Reply:
Adding a writeable check to the Mutator before saving